src/Microsoft.OData.Core/UriParser/Aggregation/AggregateTransformationNode.cs

```diff
@@ -47,5 +62,22 @@ public override TransformationNodeKind Kind
                 return TransformationNodeKind.Aggregate;
             }
         }
+
+        /// <summary>
+        /// Tries to get a <see cref="AggregateExpression"/> from it's base class.
+        /// If it fails, throws an <see cref="ODataException"/>
+        /// </summary>
+        /// <param name="expression">The object that is going to be cast.</param>
+        /// <returns>Returns a <see cref="AggregateExpression"/> or throws an <see cref="ODataException"/>.</returns>
+        private static AggregateExpression GetAggregateExpressionOrThrow(AggregateExpressionBase expression)
+        {
+            AggregateExpression aggregateExpression = expression as AggregateExpression;
+            if (aggregateExpression != null)
+            {
+                return aggregateExpression;
+            }
+
+            throw new ODataException(Strings.AggregateTransformationNode_UnsupportedAggregateExpressions());

```

Comment:
>throw new ODataException [](start = 12, length = 24)

This method is only used by the Expressions getter to  return IEnumerable<AggregateExpression>, and throwing in Getter is kind of odd. Why can't we perform the type check in the ctor instead of here?

Reply:
Cleaned that code. I think we should rename Expression collection to something more meaninful, but it will break back compat

---
In reply to: [177280247](https://github.com/OData/odata.net/pull/757#discussion_r177280247) [](ancestors = 177280247)